Transmission-gtk build failure in FreeBSD due to missing include in DetailsDialog.cc

### What is the issue?

Unable to build transmission-gtk-4.0.3 on FreeBSD due to missing #include <sys/socket.h> in DetailsDialog.cc. This occurs in a clean build environment when building for gtk3. Patch attached. 

```
[ 87% 239/264] /usr/bin/c++ -DFMT_EXCEPTIONS=0 -DFMT_HEADER_ONLY=1 -DGETTEXT_PACKAGE=\"transmission-gtk\" -DTRANSMISSIONLOCALEDIR=\"/usr/local/share/locale\" -DWITH_UTP -I/wrkdirs/usr/ports/net-p2p/transmission/work-gtk/.build/gtk -I/wrkdirs/usr/ports/net-p2p/transmission/work-gtk/transmission-4.0.3/libtransmission/.. -I/wrkdirs/usr/ports/net-p2p/transmission/work-gtk/.build/libtransmission/.. -isystem /usr/local/include/libayatana-appindicator3-0.1 -isystem /wrkdirs/usr/ports/net-p2p/transmission/work-gtk/transmission-4.0.3/cmake/../third-party/fmt/include -isystem /usr/local/include/gtkmm-3.0 -isystem /usr/local/lib/gtkmm-3.0/include -isystem /usr/local/include/atkmm-1.6 -isystem /usr/local/include/atk-1.0 -isystem /usr/local/include/glib-2.0 -isystem /usr/local/lib/glib-2.0/include -isystem /usr/local/include/glibmm-2.4 -isystem /usr/local/lib/glibmm-2.4/include -isystem /usr/local/include/sigc++-2.0 -isystem /usr/local/lib/sigc++-2.0/include -isystem /usr/local/include/giomm-2.4 -isystem /usr/local/lib/giomm-2.4/include -isystem /usr/local/include/pangomm-1.4 -isystem /usr/local/lib/pangomm-1.4/include -isystem /usr/local/include/cairomm-1.0 -isystem /usr/local/lib/cairomm-1.0/include -isystem /usr/local/include/cairo -isystem /usr/local/include/pixman-1 -isystem /usr/local/include/freetype2 -isystem /usr/local/include/libpng16 -isystem /usr/local/include/pango-1.0 -isystem /usr/local/include/harfbuzz -isystem /usr/local/include/fribidi -isystem /usr/local/include/gtk-3.0 -isystem /usr/local/include/gdk-pixbuf-2.0 -isystem /usr/local/include/gio-unix-2.0 -isystem /usr/local/include/libepoll-shim -isystem /usr/local/include/at-spi2-atk/2.0 -isystem /usr/local/include/at-spi-2.0 -isystem /usr/local/include/dbus-1.0 -isystem /usr/local/lib/dbus-1.0/include -isystem /usr/local/include/gtk-3.0/unix-print -isystem /usr/local/include/gdkmm-3.0 -isystem /usr/local/lib/gdkmm-3.0/include -isystem /usr/local/include -O2 -pipe -DLIBICONV_PLUG -fstack-protector-strong -isystem /usr/local/include -fno-strict-aliasing  -DLIBICONV_PLUG -isystem /usr/local/include -O2 -pipe -DLIBICONV_PLUG -fstack-protector-strong -isystem /usr/local/include -fno-strict-aliasing  -DLIBICONV_PLUG -isystem /usr/local/include  -DNDEBUG -std=gnu++17 -W -Wall -Wextra -Wcast-align -Wexit-time-destructors -Wextra-semi -Wextra-semi-stmt -Wextra-tokens -Wfloat-equal -Wgnu -Winit-self -Wint-in-bool-context -Wmissing-format-attribute -Wnull-dereference -Wpointer-arith -Wredundant-decls -Wredundant-move -Wreorder-ctor -Wreturn-std-move -Wself-assign -Wself-move -Wsemicolon-before-method-body -Wsentinel -Wshadow -Wsign-compare -Wsometimes-uninitialized -Wstring-conversion -Wsuggest-destructor-override -Wsuggest-override -Wuninitialized -Wunreachable-code -Wunused -Wunused-const-variable -Wunused-parameter -Wunused-result -Wwrite-strings -Wformat-security -Wno-exit-time-destructors -D_THREAD_SAFE -pthread -MD -MT gtk/CMakeFiles/transmission-gtk.dir/DetailsDialog.cc.o -MF gtk/CMakeFiles/transmission-gtk.dir/DetailsDialog.cc.o.d -o gtk/CMakeFiles/transmission-gtk.dir/DetailsDialog.cc.o -c /wrkdirs/usr/ports/net-p2p/transmission/work-gtk/transmission-4.0.3/gtk/DetailsDialog.cc
FAILED: gtk/CMakeFiles/transmission-gtk.dir/DetailsDialog.cc.o 
/usr/bin/c++ -DFMT_EXCEPTIONS=0 -DFMT_HEADER_ONLY=1 -DGETTEXT_PACKAGE=\"transmission-gtk\" -DTRANSMISSIONLOCALEDIR=\"/usr/local/share/locale\" -DWITH_UTP -I/wrkdirs/usr/ports/net-p2p/transmission/work-gtk/.build/gtk -I/wrkdirs/usr/ports/net-p2p/transmission/work-gtk/transmission-4.0.3/libtransmission/.. -I/wrkdirs/usr/ports/net-p2p/transmission/work-gtk/.build/libtransmission/.. -isystem /usr/local/include/libayatana-appindicator3-0.1 -isystem /wrkdirs/usr/ports/net-p2p/transmission/work-gtk/transmission-4.0.3/cmake/../third-party/fmt/include -isystem /usr/local/include/gtkmm-3.0 -isystem /usr/local/lib/gtkmm-3.0/include -isystem /usr/local/include/atkmm-1.6 -isystem /usr/local/include/atk-1.0 -isystem /usr/local/include/glib-2.0 -isystem /usr/local/lib/glib-2.0/include -isystem /usr/local/include/glibmm-2.4 -isystem /usr/local/lib/glibmm-2.4/include -isystem /usr/local/include/sigc++-2.0 -isystem /usr/local/lib/sigc++-2.0/include -isystem /usr/local/include/giomm-2.4 -isystem /usr/local/lib/giomm-2.4/include -isystem /usr/local/include/pangomm-1.4 -isystem /usr/local/lib/pangomm-1.4/include -isystem /usr/local/include/cairomm-1.0 -isystem /usr/local/lib/cairomm-1.0/include -isystem /usr/local/include/cairo -isystem /usr/local/include/pixman-1 -isystem /usr/local/include/freetype2 -isystem /usr/local/include/libpng16 -isystem /usr/local/include/pango-1.0 -isystem /usr/local/include/harfbuzz -isystem /usr/local/include/fribidi -isystem /usr/local/include/gtk-3.0 -isystem /usr/local/include/gdk-pixbuf-2.0 -isystem /usr/local/include/gio-unix-2.0 -isystem /usr/local/include/libepoll-shim -isystem /usr/local/include/at-spi2-atk/2.0 -isystem /usr/local/include/at-spi-2.0 -isystem /usr/local/include/dbus-1.0 -isystem /usr/local/lib/dbus-1.0/include -isystem /usr/local/include/gtk-3.0/unix-print -isystem /usr/local/include/gdkmm-3.0 -isystem /usr/local/lib/gdkmm-3.0/include -isystem /usr/local/include -O2 -pipe -DLIBICONV_PLUG -fstack-protector-strong -isystem /usr/local/include -fno-strict-aliasing  -DLIBICONV_PLUG -isystem /usr/local/include -O2 -pipe -DLIBICONV_PLUG -fstack-protector-strong -isystem /usr/local/include -fno-strict-aliasing  -DLIBICONV_PLUG -isystem /usr/local/include  -DNDEBUG -std=gnu++17 -W -Wall -Wextra -Wcast-align -Wexit-time-destructors -Wextra-semi -Wextra-semi-stmt -Wextra-tokens -Wfloat-equal -Wgnu -Winit-self -Wint-in-bool-context -Wmissing-format-attribute -Wnull-dereference -Wpointer-arith -Wredundant-decls -Wredundant-move -Wreorder-ctor -Wreturn-std-move -Wself-assign -Wself-move -Wsemicolon-before-method-body -Wsentinel -Wshadow -Wsign-compare -Wsometimes-uninitialized -Wstring-conversion -Wsuggest-destructor-override -Wsuggest-override -Wuninitialized -Wunreachable-code -Wunused -Wunused-const-variable -Wunused-parameter -Wunused-result -Wwrite-strings -Wformat-security -Wno-exit-time-destructors -D_THREAD_SAFE -pthread -MD -MT gtk/CMakeFiles/transmission-gtk.dir/DetailsDialog.cc.o -MF gtk/CMakeFiles/transmission-gtk.dir/DetailsDialog.cc.o.d -o gtk/CMakeFiles/transmission-gtk.dir/DetailsDialog.cc.o -c /wrkdirs/usr/ports/net-p2p/transmission/work-gtk/transmission-4.0.3/gtk/DetailsDialog.cc
/wrkdirs/usr/ports/net-p2p/transmission/work-gtk/transmission-4.0.3/gtk/DetailsDialog.cc:1204:42: error: use of undeclared identifier 'AF_INET'
    auto const collated_name = inet_pton(AF_INET, std::data(peer->addr), &peer_addr4) != 1 ?
                                         ^
1 error generated.
```

[patch-gtk_DetailsDialog.cc.patch](https://github.com/transmission/transmission/files/11963703/patch-gtk_DetailsDialog.cc.patch)

### Which application of Transmission?

GTK+ app on Linux, BSD, etc.

### Which version of Transmission?

4.0.3

--- a/gtk/DetailsDialog.cc
+++ b/gtk/DetailsDialog.cc
@@ -66,6 +66,7 @@
 #include <ws2tcpip.h>
 #else
 #include <arpa/inet.h>
+#include <sys/socket.h>
 #endif
 
 using namespace std::literals;

